pageserver/src/tenant/block_io.rs

```diff
@@ -84,22 +84,20 @@ where
     pub fn new(reader: R) -> Self {
         BlockCursor {
             reader,
-            cache: None,
+            cache: (u32::MAX, [0u8; PAGE_SZ]),
```

Comment:
Option should be used here, then one wouldn't have to think if u32::MAX is within block numbers or not.

How do I learn more about what this block number can be...? Feels odd how could we decide within a reader that u32::MAX is in different domain than rest of the values. Perhaps we should newtype the blocknum itself to be `0..=u32::MAX-1`?

Reply:
Block with number u32::MAX corresponds to 32Tb cache size.
Not sure that we are going to support such cache in near future:)
Th emain my motivation was to eliminate one extra check (seems to be unnoticeable) and also I have some doubts about efficient copying of data to the target array when Option is used (may be also minor)